src/sentry/grouping/__init__.py

```diff
@@ -1 +1,154 @@
+    by the SDK.  If the special `{{ default }}` value is used then this
+    variant is not used.
+
+`SaltedComponentVariant`:
+    This variant is used when the server or client produce a fingerprint
+    that refers with the special `{{ default }}` value to the default
+    grouping algorithm and the default grouping algorithm uses the
+    component based grouping system.
+
+Component Based Grouping
+------------------------
+
+The component based grouping system produces a tree of nodes (referred to
+as component) to produce the hash.  These are defined in
+`sentry.grouping.component`.  A component has an `id` which is a string
+identifying the component in the tree.  This `id` is not unique in the tree
+but unique on on one level of the hierarchy.  It has an optional `hint` which is
+a string that carries information for the user about why a value was used, not
+used or why it might have been modified.  Additionally it can carry `values`
+which are components again or primitives (strings or integers).  Lastly a
+component has a boolean `contributes` flag.  This flag when set to `False`
+removes a component (and it's children) entirely from the grouping output.
+
+Here an example of how components can be used::
+
+    function_name = 'lambda$1234'
+    threads = GroupingComponent(
+        id="function",
+        values=[function_name],
+        contributes=False,
+        hint='Unused because generated code'
+    )
+
+Strategies and Strategy Configurations
+--------------------------------------
+
+Component based grouping is triggered by strategies.  Strategies define the
+logic for how to generate a component for an interface in the event.  Each
+strategy can only produce a component for one interface.  For instance a
+stacktrace strategy can produce a component tree for a stacktrace.  Because
+events can have different forms and different strategies for the same interface
+exist, strategy configurations define which ones are picked.
+
+So for instance there is a `frame:legacy` strategy which is the legacy
+version of `frame` strategy.  Then there are the new ones (`frame:v1`,
+`frame:v2`, etc.).  Which ones are used a strategy configuration defines.
+These are in `sentry.grouping.strategies.configurations`.  A strategy can
+inherit from another one in which case a lot of behavior is inherited unless
+overridden.
+
+This for instance is how one of the configurations is defined::
+
+    register_strategy_config(
+        id="newstyle:2019-10-29",
+        base="newstyle:2019-05-08",
+        delegates=["frame:v4"],
+        risk=RISK_LEVEL_MEDIUM,
+        changelog="...",
+    )
+
+The configuration ID (`newstyle:2019-10-29`) is defined in the project
+options and then becomes the strategy configuration of choice for all new
+events.  Because in this case it inherits from another one, the default
+configuration is reused.  Here the `frame` is changed to version `v4`.
```

Comment:
This part is a little confusing. What does it mean to reuse a "default configuration"?